onlyswaps-verifier/src/config_generate.rs

```diff
@@ -1,61 +1,27 @@
-use crate::keygen::PrivateKeyMaterial;
-use crate::{AdkgPublic, AdkgSecret, GroupConfig};
 use alloy::hex;
 use alloy::primitives::FixedBytes;
+use alloy::transports::http::reqwest::Url;
 use anyhow::Context;
 use ark_bn254::G2Affine;
 use base64::Engine;
 use base64::prelude::BASE64_STANDARD;
-use clap::Parser;
+use config::adkg::{AdkgPublic, AdkgSecret, GroupConfig, PrivateKeyMaterial};
 use config::agent::AgentConfig;
-use config::app::{AppConfig, Libp2pConfig};
 use config::keys::{Bn254SecretKey, Libp2pKeyWrapper};
-use config::network::NetworkConfig;
+use config::network::{Libp2pConfig, NetworkConfig};
 use config::signing::{CommitteeConfig, MemberConfig};
 use libp2p::Multiaddr;
 use std::fs;
 use std::net::Ipv4Addr;
 use std::num::NonZeroU16;
-use std::path::PathBuf;
 use std::str::FromStr;
 use std::time::Duration;
-use url::Url;
 use utils::serialize::point::PointDeserializeCompressed;
 
-#[derive(Parser, Debug)]
-pub struct GenerateOnlyswapsConfig {
-    #[arg(
-        long = "private",
-        help = "the private key file from the generate-keys step"
-    )]
-    pub operator_private: PathBuf,
+use crate::cli::GenerateConfigArgs;
+use crate::config::AppConfig;
 
-    #[arg(long = "group", help = "the group file used to run the DKG")]
-    pub group: PathBuf,
-
-    #[arg(long = "adkg-public", help = "the ADKG public output file")]
-    pub adkg_public: PathBuf,
-
-    #[arg(long = "adkg-private", help = "the ADKG private output file")]
-    pub adkg_private: PathBuf,
-
-    #[arg(
-        long = "multiaddr",
-        help = "the multiaddr your node is running at to peer with other nodes"
-    )]
-    pub multiaddr: Multiaddr,
-
-    #[arg(
-        long = "member-id",
-        help = "the index of your node in the final committee"
-    )]
-    pub member_id: NonZeroU16,
-
-    #[arg(long, help = "the address of the router contract on each chain")]
-    pub router_address: Option<FixedBytes<20>>,
-}
-
-pub(crate) fn generate_onlyswaps_config(args: GenerateOnlyswapsConfig) -> anyhow::Result<()> {
+pub(crate) fn generate_onlyswaps_config(args: GenerateConfigArgs) -> anyhow::Result<()> {
```

Comment:
i've just realized that you use `t` from the group file when building the onlyswaps config. You should instead set the onlyswaps config threshold to `t = t_reconstruction + 1`

Reply:
ah good shout, have fixed this